libraries/rush-lib/src/api/CustomTipsConfiguration.ts

```diff
@@ -0,0 +1,83 @@
+import { ConsoleTerminalProvider, ITerminal, JsonFile, Terminal } from '@rushstack/node-core-library';
+
+export interface IRushCustomTipsJson {
+  prefix?: string;
+  customTips?: ICustomTipItemJson[];
+}
+
+/** Similar to {@link TerminalProviderSeverity} but without "debug" and "verbose" */
+export enum TipSeverity {
+  log = 0,
+  warning = 1,
+  error = 2
+}
+
+export interface ICustomTipItemJson {
+  id: CustomTipId;
+  tip: string;
+  prefix?: string;
+  severity?: TipSeverity;
+}
+
+// TODO: consider making this work with the plugin (e.g., the plugins are able to define their own customizable tips)
+export type CustomTipId = 'PNPM_MISMATCH_DEPENDENCY' | 'ANOTHER_ERROR_ID';
+
+export class CustomTipsConfiguration {
+  private _config: IRushCustomTipsJson;
+  private _tipMap: Map<CustomTipId, ICustomTipItemJson>;
+  private _terminal: ITerminal;
+
+  public constructor(configFilename: string) {
+    this._tipMap = new Map();
+    this._config = {};
+    this._terminal = new Terminal(new ConsoleTerminalProvider());
```

Comment:
Ideally we pass in the terminal from the calling action. Probably best to take it as a parameter to `.log`

Reply:
That makes sense. It's done.